### PR TITLE
SorrTask687 Improve all.development_documents.reference.md

### PR DESCRIPTION
**Pull request by nehalR777**
@samarth9008 @gpsaggese , I have Fixed broken links in the documentation and added the codebase clean-up doc. I have also removed the references to dataflow , datapull and HTMLcov as requested here https://github.com/sorrentum/sorrentum/issues/687#issuecomment-1965681912

### CODE REVIEW

**Comment by nehalR777** [comment]
@gpsaggese sir , awaiting your review and approval on this. 
cc: @samarth9008 @sonaalKant

File: docs/onboarding/all.development_documents.reference.md [hunk 1]
```diff
@@ -83,7 +82,7 @@ This contains a set of resources that over time will make 10x more productive
 
 - [Buildmeister process](/docs/work_organization/all.buildmeister.how_to_guide.md)
 
-- [HTMLcov server](/docs/infra/ck.use_htmlcov_server.how_to_guide.md)
+- HTMLcov server
```
Comment (samarth9008) [inline docs/onboarding/all.development_documents.reference.md:85]: Lets delete if no link is there. Same for datapull and dataflow

**Review by samarth9008 — CHANGES_REQUESTED** [review]
Good work! Small change.

File: docs/onboarding/all.development_documents.reference.md (hunk 1 above)
Comment (nehalR777) [reply docs/onboarding/all.development_documents.reference.md]: @samarth9008 , the changes have been made.

**Comment by nehalR777** [comment]
@samarth9008 , the changes have been made as requested , please review

**Pull request merged**